### ISSUE TITLE
Properly display generics in Markdown

### ISSUE THREAD

**Issue by yiliansource** [tgreyuk/typedoc-plugin-markdown#239]
I'm so sorry for having [so many issues](https://github.com/tgreyuk/typedoc-plugin-markdown/issues/215) with generics in TypeDoc. 😅

The Markdown renderer seems to have issues properly representing and linking generic types inside the rendered result. At the moment, the class `Spline<Color>` will be rendered as `Spline`<[Color]()\>. Sadly, it does not seem possible to provide links inside of code snippets, so `Spline<[Color](...)>` would not work. The only solution I'd see at the moment for generics is to provide HTML code directly with something along

```html
<code>Spline&lt;<a href="">Color</a>&gt;</code>
```

which renders as <code>Spline&lt;<a href="">Color</a>&gt;</code>.

Is this something that can feasibly be fixed, or might this even be intended/unfixable behaviour?

**Comment by tgreyuk** [comment]
Just to clarify that this use-case is for manually adding references inside doc comments?

**Comment by yiliansource** [comment]
Sorry, should have properly clarified and not rushed before work.

The use case is that generic types don't seem to be properly rendered. I don't believe this is related to doc comments. I'll add example code and the current result:

---

```ts
/**
 * Represents a spline that can take numeric values.
 */
export class NumericSpline extends Spline<number> {
    // ...
}
```

# Class: NumericSpline

[components](../modules/components.md).NumericSpline

Represents a spline that can take numeric values.

## Hierarchy

- `Spline`<number\>

  ↳ **NumericSpline**

---

As you can see, in the _hierarchy_ section, the full name of the variable is not wrapped in the `<code>` tags, causing the "generic" brackets to feel out of place.

I'm happy to elaborate more, if needed.

**Comment by tgreyuk** [comment]
Ah sorry - ok i see what you are saying. I will have a proper look at this and report back.

**Comment by yiliansource** [comment]
I'd be glad to take a peek aswell if you like, I'd be happy to contribute - I just don't really know how the codebase works at the moment. If you'd rather someone else bothers with this, feel free to give me a hint in the right direction!

**Comment by tgreyuk** [comment]
Like you suggest this has been implemented like this as its not possible to nest links inside backticks in markdown and trying to avoid html in the output.

What do you think about adding each symbol in a backtick rather than the entire string? It means the symbols are consistent although maybe the angle brackets look out of place?

```
- [`Spline`](spline.md)<`number`, [`Color`](../color.md)\>
``` 

<img width="252" alt="Screenshot 2021-06-23 at 13 36 43" src="https://user-images.githubusercontent.com/11680870/123097884-744e4900-d428-11eb-8f51-e9d7a2e4a269.png">

Any other suggestions welcome?

> I'd be glad to take a peek aswell if you like, I'd be happy to contribute 

I am happy to pick this up, but yes contributions very welcome. Actually been planning on writing a contributing page but currently in the middle of refactoring to remove handlebars so got a bit sidetracked.

**Comment by yiliansource** [comment]
Encapsulating each symbol in backticks is definitely an improvement - but depending on the CSS used to style the rendered Markdown, the angle brackets may look pretty out of place. I get how one would want to avoid HTML in the output though. I'd say before making that kind of concession most people would take the slightly out-of-place brackets and commas any day.

> I am happy to pick this up, but yes contributions very welcome. Actually been planning on writing a contributing page but currently in the middle of refactoring to remove handlebars so got a bit sidetracked.

I'll leave you to do the magic then! Am definitely excited for the contributing page though, but obviously don't rush yourself.

**Comment by tgreyuk** [comment]
ok have implemented in typedoc-plugin-markdown@3.10.1. Thanks I think it's an improvement. Let me know if you spot anything.

**Comment by yiliansource** [comment]
Perfect, looks good to me! Thanks for the quick fix!
